101-azure-machine-learning-service-workspace/azuredeploy.json

```diff
@@ -0,0 +1,123 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "workspaceName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the Azure Machine Learning service workspace."
+            }
+        },
+        "location": {
+            "type": "string",
+            "allowedValues": [
+                "East US",
+                "East US 2",
+                "South Central US",
+                "Southeast Asia",
+                "West Central US",
+                "West Europe",
+                "West US 2"
+            ],
+            "metadata": {
+                "description": "The location where the workspace will be created."
+            }
+        }
+     },
+    "variables": {
+        "storageAccount": {
+            "name": "[concat('sa',uniqueString(resourceGroup().id))]",
+            "type": "Standard_LRS"
+        },
+        "keyVault": {
+            "name": "[concat('kv',uniqueString(resourceGroup().id))]",
+            "tenantId": "[subscription().tenantId]"
+        },
+        "applicationInsightsName": "[concat('ai',uniqueString(resourceGroup().id))]",
+        "containerRegistryName": "[concat('cr',uniqueString(resourceGroup().id))]"
+    },
+    "resources": [
+        {
+            "name": "[variables('storageAccount').name]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "location": "[parameters('location')]",
+            "apiVersion": "2018-07-01",
+            "sku": {
+                "name": "[variables('storageAccount').type]"
+            },
+            "kind": "StorageV2",
+            "properties": {
+                "encryption": {
+                    "services": {
+                        "blob": {
+                            "enabled": true
+                        },
+                        "file": {
+                            "enabled": true
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                },
+                "supportHttpsTrafficOnly": true
+            }
+        },
+        {
+            "name": "[variables('keyVault').name]",
+            "type": "Microsoft.KeyVault/vaults",
+            "apiVersion": "2018-02-14",
+            "location": "[parameters('location')]",
+            "properties": {
+                "tenantId": "[variables('keyVault').tenantId]",
+                "sku": {
+                    "name": "standard",
+                    "family": "A"
+                },
+                "accessPolicies": []
+            }
+        },
+        {
+            "name": "[variables('applicationInsightsName')]",
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2015-05-01",
+            "location": "[parameters('location')]",
```

Comment:
This needs to be restricted to certain locations - will need another param if it doesn't match your current restriction

Reply:
These should match; when you create through the portal everything gets created in the same locations. What's supported by Azure Machine Learning service should be the intersection of what all the other dependencies allow.

Reply:
What about an IF statement? There's only two locations that aren't available, east us 2 and west central us. Using an IF, I can split the difference and select south central us for both of those for the app insight resource.